amundsen_application/static/js/ducks/issue/reducer.ts

```diff
@@ -0,0 +1,111 @@
+import { Issue } from "interfaces";
+import { 
+  GetIssues, 
+  CreateIssue, 
+  GetIssuesResponse, 
+  CreateIssueRequest,
+  GetIssuesRequest,
+  CreateIssueResponse} 
+  from './types'; 
+
+
+/* ACTIONS */
+export function createIssue(formData: FormData): CreateIssueRequest {
+  return {
+    payload: {
+      data: formData,
+    },
+    type: CreateIssue.REQUEST,
+    };
+};
+
+export function createIssueSuccess(issue: Issue): CreateIssueResponse {
+  return {
+    type: CreateIssue.SUCCESS, 
+    payload: {
+      issue
+    }
+  };
+};
+
+export function createIssueFailure(): CreateIssueResponse {
+  return {
+    type: CreateIssue.FAILURE, 
+    payload: {
+      issue: null
+    }
+  };
+};
+
+export function getIssues(tableKey: string): GetIssuesRequest {
+  return {
+    type: GetIssues.REQUEST, 
+    payload: {
+      key: tableKey
+    }
+  }; 
+}
+
+export function getIssuesSuccess(issues: Issue[], remainingIssues?: number, remainingIssuesUrl?: string): GetIssuesResponse {
+  return { 
+    type: GetIssues.SUCCESS, 
+    payload: {
+      issues, 
+      remainingIssues, 
+      remainingIssuesUrl
+    }
+  }
+}
+
+export function getIssuesFailure(): GetIssuesResponse {
+  return { 
+    type: GetIssues.FAILURE, 
+    payload: {
+      issues: [], 
+      remainingIssues: 0,
+      remainingIssuesUrl: null
+    }
+  }
+}
```

Comment:
There are many places where we can set the default values on failures, but we previously decided it should be in the sagas because that allows for the case where we might want to transform the response from the api call before it gets passed back to the application. Suggested change would have to be made together with other items suggested below.
```
export function getIssuesFailure(issues: Issue[], remainingIssues?: number, remainingIssuesUrl?: string): GetIssuesResponse {
  return { 
    type: GetIssues.FAILURE, 
    payload: {
      issues, 
      remainingIssues,
      remainingIssuesUrl
    }
  }
}
```

Reply:
Edit: Just return `initialIssueState` and this action won't need a payload unless we implement special logic for teh failure case.